tests/AssetManagerTest/Service/AssetManagerTest.php

```diff
@@ -208,7 +211,53 @@ public function testSetExtensionFilters()
         $assetManager->setAssetOnResponse($response);
         $this->assertEquals($minified, $response->getBody());
     }
+    public function testSetExtensionFiltersNotDuplicate()
+    {
+        $config = array(
+            'filters' => array(
+                'js' => array(
+                    array(
+                        'filter' => '\ReverseFilter',
+                    ),
+                ),
+            ),
+        );
+        
+        $assetFilterManager = new AssetFilterManager($config['filters']);
+        $mimeResolver       = new MimeResolver;
+        $assetFilterManager->setMimeResolver($mimeResolver);
+
+        $aggregateResolver  = new AggregateResolver;        
+        $mockedResolver     = $this->getResolver(__DIR__ . '/../../_files/require-jquery.js');
+        $collArr = array(
+            'blah.js' => array(
+                'asset-path'
+            )
+        );
+        $resolver = new CollectionResolver($collArr);
+        $resolver->setAssetFilterManager($assetFilterManager);
+        $resolver->setAggregateResolver($aggregateResolver);
+
+        $aggregateResolver->attach($mockedResolver, 500);
+        $aggregateResolver->attach($resolver, 1000);
 
+        
+        $response     = new Response;
+        $request      = $this->getRequest();
+        // Have to change uri because asset-path would cause an infinite loop
+        $request->setUri('http://localhost/base-path/blah.js');
+        
+        $assetCacheManager = $this->getAssetCacheManagerMock();
+        $assetManager      = new AssetManager($aggregateResolver, $config);
+        $assetManager->setAssetCacheManager($assetCacheManager);
+        $assetManager->setAssetFilterManager($assetFilterManager);
+
+        $this->assertTrue($assetManager->resolvesToAsset($request));
+        $assetManager->setAssetOnResponse($response);
+
+        $reversedOnlyOnce = '1' . strrev(file_get_contents(__DIR__ . '/../../_files/require-jquery.js'));
+        $this->assertEquals($reversedOnlyOnce, $response->getBody());
```

Comment:
I can do that, it will be much easier. But I thought you would like a test that probes that fails with the code *as it is* (before the $filterInstance array).

It would be very easy to create a test in AssetFilterManager that checks that an instance is not instantiated more than once. 

But I was sure once we talk you said "provide a failing test" so that's why I created this test.

If you are already convinced of the problem this test probes, I can delete it, and create on in AssetFilterTest to check the instances.